source/CIS/cis-4.rst

```diff
@@ -0,0 +1,705 @@
+
+``issuer``
+^^^^^^^^^^
+
+Query the issuer's account address.
+
+Response
+~~~~~~~~
+
+The function output is the issuer's account address.
+It is serialized as :ref:`CIS-4-AccountAddress`.
+
+.. _CIS-4-functions-issuerMetadata:
+
+``issuerMetadata``
+^^^^^^^^^^^^^^^^^^
+
+Query the current token metadata URLs for a list of token IDs.
+
+Response
+~~~~~~~~
+
+The function output is the issuer's metadata URL.
+
+It is serialized as :ref:`CIS-2-MetadataUrl`.
+
+.. _CIS-4-functions-registerCredential:
+
+``registerCredential``
+^^^^^^^^^^^^^^^^^^^^^^
+
+Register a credential with the given ID.
+
+Parameter
+~~~~~~~~~
+
+The parameter is the credential ID and credential information that is used to create an entry in the registry.
+
+It is serialized as :ref:`CIS-4-CredentialID` (``credential_id``) followed by :ref:`CIS-4-CredentialInfo` (``credential_info``)::
+
+  RegisterCredentialParameter ::= (credential_id: CredentialID) (credential_info: CredentialInfo)
+
+Requirements
+~~~~~~~~~~~~
+
+- The credential registration request MUST fail if the credential ID is already present in the registry.
+- The credential status after registration MUST NOT be ``Revoked`` (See the possible values for the status in :ref:`CIS-4-CredentialStatus`).
+
+.. _CIS-4-functions-revokeCredentialIssuer:
+
+``revokeCredentialIssuer``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Revoke a credential by the issuer's request.
+The issuer is authorized to revoke a credential if the transaction sender's address is the same as the return value of :ref:`CIS-4-functions-issuer`.
+
+Parameter
+~~~~~~~~~
+
+The parameter is the credential ID :ref:`CIS-4-CredentialID` and optional string indicating the revocation reason.
+
+It is serialized as :ref:`CIS-4-CredentialID` followed by 1 byte to indicate whether a reason is included, if its value is 0, then no reason string present, if the value is 1 then the bytes corresponding to the reason string follow::
+
+  OptionReason ::= (0: Byte)
+                 | (1: Byte) (n: Byte) (reason_string: Byteⁿ)
+  RevokeCredentialIssuerParam ::= (credential_id: CredentialID) (reason: OptionReason)
+
+.. TODO: what kind of characters are allowed? ASCII, Unicode?
+
+Requirements
+~~~~~~~~~~~~
+
+- If revoked successfully, the credential status MUST change to ``Revoked`` (see :ref:`CIS-4-functions-credentialStatus`).
+- The revocation MUST fail if:
+    - The the transaction ``sender`` is not the issuer.
+    - The credential ID is not present in the registry.
+    - The credential status is not one of ``Active`` or ``NotActivated`` (see :ref:`CIS-4-functions-credentialStatus`).
+
+.. _CIS-4-functions-revokeCredentialHolder:
+
+``revokeCredentialHolder``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Revoke a credential by the holders's request.
+
+The holder is authorized to revoke a credential by verifying the signature with the holder's public key.
+It replaces the authorization checks conducted on the ``sender/invoker`` variable with signature verification.
```

Comment:
What is the `sender/invoker` variable? Are you referring to the sender of the contract update? If so, consider making this more explicit, ideally this standard would make sense to people who are not familiar to our smart contract Rust API